sys/geom/geom_ctl.h

```diff
@@ -62,18 +73,35 @@ struct gctl_req_arg {
 #define GCTL_PARAM_VALUEKERNEL	16
 #define GCTL_PARAM_CHANGED	32
 
+#ifdef _KERNEL
+/*
+ * XXX: This should really be gctl_kreq, but that would require a
+ * larger sweep of renames in the kernel.
+ */
 struct gctl_req {
+	u_int				narg;
+	struct gctl_kreq_arg		*arg;
+	int				nerror;
+	struct sbuf			*serror;
+};
+#endif
+
+#ifdef _KERNEL
+struct gctl_ureq {
```

Comment:
I'm confused why all this needs to exist. With CheriABI as the native ABI, we shouldn't need to convert from user to kernel, only in the freebsd64 code?